pandas/core/arrays/boolean.py

```diff
@@ -314,6 +314,20 @@ def __getitem__(self, item):
             return self._data[item]
         return type(self)(self._data[item], self._mask[item])
 
+    def __eq__(self, other):
+        if not isinstance(other, BooleanArray):
+            return NotImplemented
+        return (
+            hasattr(other, "_data")
```

Comment:
BooleanArray should already have a `__eq__` method (see `create_comparison_method` Was this needed?

Reply:
Not needed. Missed that in my WIP PR.